geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/filters/Z2Filter.scala

```diff
@@ -38,10 +39,16 @@ class Z2Filter(val xy: Array[Array[Int]]) extends RowFilter {
 
 object Z2Filter extends RowFilterFactory[Z2Filter] {
 
+  import org.locationtech.geomesa.index.conf.QueryHints.RichHints
+
   private val RangeSeparator = ":"
   private val TermSeparator  = ";"
 
-  def apply(values: Z2IndexValues): Z2Filter = {
+  def apply(values: Z2IndexValues): Z2Filter = apply(values, seek = true)
+
+  def apply(values: Z2IndexValues, hints: Hints): Z2Filter = apply(values, hints.isScanSeeking)
+
+  private def apply(values: Z2IndexValues, seek: Boolean): Z2Filter = {
```

Comment:
Ok, this is just wiring up the hint and not using it, right?  Do we want to set up seeking for Z2?  If not, should we be clear about the difference in the docs?  (That is, that one can only seek with Z3 initially?)

Reply:
we should probably set up seeking for z2 eventually, but I haven't done it yet. i don't think we need to get into the specifics in the docs, given that this ideally should be transparent to users and is configurable mainly as a hedge against regression.

Reply:
+1 +1 + 1